src/nvim/msgpack_rpc/server.c

```diff
@@ -224,7 +224,7 @@ int server_start(const char *endpoint)
 /// @param endpoint Address of the server.
 void server_stop(char *endpoint)
 {
-  Server *server;
+  Server *server = NULL;
```

Comment:
This one is a bug, in fact. One that would never occur, but a bug after all. Problem is in:
```
  int i = 0;  // The index of the server whose address equals addr.
  for (; i < servers.ga_len; i++) {
    server = ((Server **)servers.ga_data)[i];
    if (strcmp(addr, server->addr) == 0) {
      break;
    }
  }

  if (i == servers.ga_len) {
    ELOG("Not listening on %s", addr);
    return;
  }
```
This code wouldn't work for server.ga_len < 0.
We know that won't happen, but from the compiler point of view, it's an int, and it could.
That case leads to function not exiting at ELOG block, and server being used without initialization.
It can all be fixed (without initializing variable) by changing it into:
```
  int i = -1;  // The index of the server whose address equals addr.                                                                                  
   for (int j = 0; j < servers.ga_len; j++) {                                                                                                          
     server = ((Server **)servers.ga_data)[j];                                                                                                         
     if (strcmp(addr, server->addr) == 0) {                                                                                                            
       i = j;                                                                                                                                          
       break;                                                                                                                                          
     }                                                                                                                                                 
   }                                                                                                                                                   
                                                                                                                                                       
   if (i == -1) {                                                                                                                                      
     ELOG("Not listening on %s", addr);                                                                                                                
     return;                                                                                                                                           
   }
```
Or, alternatively, introducing a server_found bool variable.
This is an example of how leaving variables uninitialized where it makes sense can actually help.